### PR TITLE
Made some improvements proposed from the GA code audit

### PR DESCRIPTION
**Pull request by roy-powerfinance**
*  Added comments to burn() and issue()
* Added comments and weights to functions in the GA module
* Added zero checks to some functions
* Removed some repetitive code

### CODE REVIEW

File: frame/generic-asset/src/lib.rs [hunk 1]
```diff
@@ -382,15 +391,44 @@ decl_module! {
...
+		/// of the transfer, the account will be reaped.
+		///
+		/// The dispatch origin for this call must be `Signed` by the transactor.
+		///
+		/// # <weight>
+		/// - Dependent on arguments but not critical, given proper implementations for
+		///   input config types. See related functions below.
+		/// - It contains a limited number of reads and writes internally and no complex computation.
+		///
+		/// Related functions:
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: Not sure 100% of this comment applies to GA, would be good to remove those parts that don't

File: frame/generic-asset/src/lib.rs [hunk 2]
```diff
@@ -411,26 +449,39 @@ decl_module! {
 			}
 		}
 
-		/// Mints an asset, increases its total issuance.
+		/// Mints an asset, increases its total issuance. Deposits the newly mint currency into target account
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: `mint` -> `minted`

File: frame/generic-asset/src/lib.rs [hunk 3]
```diff
@@ -411,26 +449,39 @@ decl_module! {
...
+		#[weight = SimpleDispatchInfo::FixedNormal(500_000)]
 		fn mint(origin, #[compact] asset_id: T::AssetId, to: T::AccountId, amount: T::Balance) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 			Self::mint_free(&asset_id, &who, &to, &amount)?;
 			Self::deposit_event(RawEvent::Minted(asset_id, to, amount));
 			Ok(())
 		}
 
-		/// Burns an asset, decreases its total issuance.
+		/// Burns an asset, decreases its total issuance. Deduct the money from target account
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs:456]: would be good to specify target is \`to\`

File: frame/generic-asset/src/lib.rs [hunk 4]
```diff
@@ -411,26 +449,39 @@ decl_module! {
...
+		#[weight = SimpleDispatchInfo::FixedNormal(500_000)]
 		fn burn(origin, #[compact] asset_id: T::AssetId, to: T::AccountId, amount: T::Balance) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 			Self::burn_free(&asset_id, &who, &to, &amount)?;
 			Self::deposit_event(RawEvent::Burned(asset_id, to, amount));
 			Ok(())
 		}
 
-		/// Can be used to create reserved tokens.
+		/// Can be used to create a new asset with reserved asset_id.
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: ~Can be used to~ Create a new asset with a reserved \`asset_id\`.
\`Internally calls \`create_asset\` with the given \`asset_id\`

File: frame/generic-asset/src/lib.rs [hunk 5]
```diff
@@ -370,8 +372,15 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		/// Create a new kind of asset and nominates the owner of this asset. The
+		/// Create a new kind of asset and nominates the owner of this asset.
+		/// The asset_id will be the next unoccupied asset_id
+		/// Accounts who will have the permissions to min/burn/change permission are passed in via 'options'
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: `min/burn/change` -> min**t**

**Review by jordy25519 — COMMENTED** [review]
Looks good, comments need some small tidy ups

File: frame/generic-asset/src/lib.rs (hunk 1 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs]: Updated

File: frame/generic-asset/src/lib.rs (hunk 3 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs:456]: Updated

File: frame/generic-asset/src/lib.rs (hunk 4 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs]: Updated

File: frame/generic-asset/src/lib.rs (hunk 5 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs]: Updated

File: frame/generic-asset/src/lib.rs (hunk 2 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs]: Updated

File: frame/generic-asset/src/lib.rs [hunk 6]
```diff
@@ -1348,11 +1390,9 @@ where
 	}
 
 	fn slash_reserved(who: &T::AccountId, value: Self::Balance) -> (Self::NegativeImbalance, Self::Balance) {
-		let b = Self::reserved_balance(&who.clone());
-		let slash = cmp::min(b, value);
-
-		<Module<T>>::set_reserved_balance(&U::asset_id(), who, b - slash);
-		(NegativeImbalance::new(slash, U::asset_id()), value - slash)
+		//slash_reserved(asset_id: &T::AssetId, who: &T::AccountId, amount: T::Balance) -> Option<T::Balance> {
```
Comment (mzxyz) [inline frame/generic-asset/src/lib.rs]: Remove this comment

**Review by mzxyz — APPROVED** [review]

**Review by jordy25519 — APPROVED** [review]
lgtm!

File: frame/generic-asset/src/lib.rs (hunk 6 above)
Comment (roy-powerfinance) [reply frame/generic-asset/src/lib.rs]: got it thank you!

**Pull request merged**